pkg/queryfrontend/labels_codec.go

```diff
@@ -87,8 +87,25 @@ func (c labelsCodec) MergeResponse(responses ...queryrange.Response) (queryrange
 		if len(responses) == 1 {
 			return responses[0], nil
 		}
-		seriesData := make(labelpb.ZLabelSets, 0)
 
+		i := 0
+		var resp queryrange.Response
+		for _, response := range responses {
+			if len(response.(*ThanosSeriesResponse).Data) > 0 {
```

Comment:
I don't get this, why we cannot check this in loop below?

Reply:
Makes sense. As you mentioned I ll remove this change since it is unrelated and fix them in next master